### PR TITLE
[usr_only] compensate for corner cases

### PR DESCRIPTION
**Pull request by 257**
i kept hitting this with OutputFormat.gpt_btrfs.
maybe the logic of including usr_only in minimize cases requires
revisiting? maybe "/usr" itself can be a subvol?

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
@poettering Can you also check if these changes are OK?

File: mkosi/__init__.py [hunk 1]
```diff
@@ -891,7 +891,7 @@ def determine_partition_table(args: CommandLineArguments) -> Tuple[str, bool]:
             pn += 1
             run_sfdisk = True
 
-    if not is_generated_root(args):
+    if not is_generated_root(args) and not args.usr_only:
```
Comment (poettering) [inline mkosi/__init__.py:894]: is_generated_root() returns true for args.usr_only already, this change should not alter anything hence?

File: mkosi/__init__.py [hunk 2]
```diff
@@ -1616,12 +1616,13 @@ def prepare_tree(args: CommandLineArguments, root: str, do_run_build_script: boo
         if args.output_format is OutputFormat.subvolume or (
             args.output_format is OutputFormat.gpt_btrfs and not args.minimize
         ):
-            btrfs_subvol_create(os.path.join(root, "home"))
-            btrfs_subvol_create(os.path.join(root, "srv"))
-            btrfs_subvol_create(os.path.join(root, "var"))
-            btrfs_subvol_create(os.path.join(root, "var/tmp"), 0o1777)
-            os.mkdir(os.path.join(root, "var/lib"))
-            btrfs_subvol_create(os.path.join(root, "var/lib/machines"), 0o700)
+            if not args.usr_only:
```
Comment (poettering) [inline mkosi/__init__.py:1619]: maybe this also should check for is_generated_root()?

**Review by poettering — CHANGES_REQUESTED** [review]

File: mkosi/__init__.py (hunk 2 above)
Comment (257) [reply mkosi/__init__.py]: maybe? i changed it to ```is_generated_root()``` include all cases

File: mkosi/__init__.py (hunk 1 above)
Comment (257) [reply mkosi/__init__.py]: Oops!

**Review by DaanDeMeyer — APPROVED** [review]

**Pull request merged**